scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,635 @@
+    "
+        fi
 
-    for i in .bash_profile .bash_login .profile; do
-        fn="$HOME/$i"
-        if [ -w "$fn" ]; then
-            if ! grep -q "$p" "$fn"; then
-                echo "modifying $fn..." >&2
-                echo "if [ -e $p ]; then . $p; fi # added by Nix installer" >> "$fn"
+    fi
+
+    if [ -d "$dest" ]; then
+        print "Directory $dest already exists. Skipping creation of $dest"
+    fi
+
+}
+
+installNix() {
+
+    # In this function, if some action makes any change to the system, or does err Exit - it's need to be wrapped in a function, ex func(){...}. Inside func() you start with setup of traps, before errors/actions happen, so and according revert function be launched: 'trap funcRevertOnCondition condition', and funcRevertOnCondition is described right after trap.
+    # How all actions operate inside this function:
+    #**************************************************
+    #
+    #    installNix() {
+    #
+    #        func() {
+    #
+    #            trap 'funcRevertOnCondition $LINENO $?' condition
+    #            funcRevertOnCondition() {
+    #                errorRevert "Received Error signal from line $1"
+    #                # revert action
+    #                >&2 contactme # This function gives contact information
+    #                exit "$?"
+    #            }
+    #
+    #            # action of functions
+    #            # 'exit SIG # to the trap that catches exit SIG'
+    #        }
+    #
+    #        func
+    #
+    #    }
+    #
+    #**************************************************
+
+    # Function must have revert procedure. In case of any problem - function quits without artifacts/changes to the system
+    # Nix is pure, and claims to be transactional, so let's show people clean transactional installation
+
+    createDest() {
+
+        trap destCreateRevert 1
+
+        destCreateRevert() {
+            # all means failed, Exit!
+            error "
+
+    Can't get root access to create $dest folder. All variants failed.
+    Exiting...
+    "
+        }
+
+        # Shell parses variable with current user, so value under sudo/su not changes.
+        # It is const, but don't move it above USER set/checks.
+        readonly cmd="mkdir -m 0755 $dest && chown $USER $dest"
+
+        # Trying to create $dest by all means
+
+        # For case user is root, or has access to create $cmd. Else -> do 'then' section
+        if sh -c "$cmd" > /dev/null 2>&1 ; then
+            print "Creating $dest directory by running $cmd using $USER rights"
+        else
+            if command -v sudo > /dev/null 2>&1 ; then
+                print "Creating $dest directory by running $cmd using sudo:"
+                if sudo -n true 2> /dev/null ; then
+                    print "Checked: sudo access does not require password."
+                else
+                    print "Checked: sudo access requires to enter password:"
+                fi
```

Comment:
Probably just delet this last if/else, it doesn't do much useful for the user I think?

Reply:
Yes. That is probably too much info.
And it does not have sense to leave a half of the check, it is dualistic.
So all `if-else-fi` can be removed.

Since anyway `sudo` can has a time frame of being passwordless.
And message passes without a pause.